internal/charm/config.go

```diff
@@ -49,21 +47,15 @@ func (c secretC) Coerce(v interface{}, path []string) (interface{}, error) {
 	if err != nil {
 		return nil, err
 	}
-	str := s.(string)
-	if str == "" {
+	uriStr := s.(string)
+	if uriStr == "" {
 		return "", nil
 	}
-	u, err := url.Parse(str)
+	uri, err := secrets.ParseURI(uriStr)
```

Comment:
This is a very slight change, ParseURI behaves slightly differently

And note that we return the stringified secret URI as the coerced value

Reply:
Why did we make the change?

Reply:
Flyby improvement

Reply:
Yeah drop it.

Reply:
Done